src/precompile.cuh

```diff
@@ -133,7 +134,40 @@ namespace precompile_operations {
             error_code = ERR_RETURN;
         }
     }
+
+  __host__ __device__ __forceinline__ static void operation_RIPEMD160(arith_t &arith,
+                                                                      bn_t &gas_limit,
+                                                                      bn_t &gas_used,
+                                                                      uint32_t &error_code,
+                                                                      return_data_t &return_data,
+                                                                      message_t &message
+                                                                      )
+  {
+    cgbn_add_ui32(arith._env, gas_used, gas_used, GAS_PRECOMPILE_RIPEMD160);
+
+
+    size_t size;
+    uint8_t *data;
+
+    uint8_t *input;
+    size = message._content->data.size;
+    input = message._content->data.data;
+
+    uint8_t hash[20];
+    uint8_t output[32] = {0};
+
+    bn_t length;
+    cgbn_set_ui32(arith._env, length, size);
+    arith.evm_words_gas_cost(gas_used, length, GAS_PRECOMPILE_RIPEMD160_WORD);
```

Comment:
You can create a similar function to keccak and sha256 in the arith class for the gas cost, with the idea that the cost function will change in the future.